ipatests/test_integration/test_sssd.py

```diff
@@ -148,3 +149,38 @@ def test_is_user_filtered(self, user):
             dp_req = ("Looking up [{0}] in data provider".format(
                 self.users[user]['name']))
             assert not dp_req.encode() in sssd_log
+
+    def test_ext_grp_with_ldap(self):
+        """IPA objects should read via LDAP when resolving trusted users.
+
+        Regression test for https://pagure.io/SSSD/sssd/issue/4073
+
+        When aduser is added in extrnal group and this is group is added
+        in group with same name of nonprivate ipa user and possix id, then
+        lookup of aduser and group should be successful when cache is empty.
+        """
+        client = self.clients[0]
+        user = 'ipatest'
+        id = '100996'
+        ext_group = 'ext-ipatest'
+        tasks.kinit_admin(self.master)
+        tasks.user_add(self.master, user, extra_args=[
+            '--noprivate', '--uid', id, '--gidnumber', id])
+        tasks.group_add(self.master, user, extra_args=['--gid', id])
+        tasks.group_add(self.master, ext_group, extra_args=['--external'])
+        self.master.run_command(
+            ['ipa', 'group-add-member', '--group', ext_group, user])
+        self.master.run_command([
+            'ipa', 'group-add-member', '--external',
+            self.users['ad']['name'], ext_group,
+            '--users=', '--groups='])
+        self.master.run_command(['sss_cache', '-E'])
+        tasks.clear_sssd_cache(self.master)
+        client.run_command(['sss_cache', '-E'])
+        tasks.clear_sssd_cache(client)
+        try:
+            result = client.run_command(['id', self.users['ad']['name']])
+            assert id and user in result.stdout_text
+        finally:
+            self.master.run_command(['ipa', 'user-del', user])
+            self.master.run_command(['ipa', 'group-del', user, ext_group])
```

Comment:
The test creates two groups but cleanup removes only one

Reply:
It adds two groups : user, ext_group
        # add group with same name as user and id.
        tasks.group_add(self.master, user, extra_args=['--gid', id])
        tasks.group_add(self.master, ext_group, extra_args=['--external'])

Reply:
ah, sorry, did not notice that you delete two groups in one command